### ISSUE TITLE
Double activation of button does not correctly fire central scene event

### ISSUE THREAD

**Issue by JonasPed** [OpenZWave/Zwave2Mqtt#241]
**Version**

Build/Run method
- [x] Docker
- [ ] PKG
- [ ] Manually built (git clone - npm install - npm run build ) 

Zwave2Mqtt version: 2.1.1
Openzwave Version: 1.6.0

**Describe the bug**
Double click button on LogicHome ZHC5010 not published correctly to MQTT.

**To Reproduce**
1. Double click on any of the four buttons on a LogicHome ZHC5010 switch. 
2. This should send a central scene event with scene id for the button double activated and an key attribute of 3. 
3. Verify messages published to MQTT. Only a single message is published for command class 91 and the value is Inactive. 

**Expected behavior**
I would expect two messages published to MQTT. One with value "double click activated" and one with value "Inactive". 

**Additional context**
The file zwave2mqtt.txt contains output from zwave2mqtt docker container. As can be seen in line 21 OpenZwave correctly sees the double activation. 

The file mqtt_message.tx contains the message published to MQTT. 
[mqtt_message.txt](https://github.com/OpenZWave/Zwave2Mqtt/files/4162066/mqtt_message.txt)
[zwave2mqtt.txt](https://github.com/OpenZWave/Zwave2Mqtt/files/4162067/zwave2mqtt.txt)

**Comment by robertsLando** [comment]
@Fishwaldo

**Comment by Fishwaldo** [comment]
Nothing wrong on OZW side: zwave2mqtt_1  | 2020-02-05 20:46:54.394 Info, Node049, Received Central Scene set from node 49: scene id=4 with key Attribute 3. Sending event notification.

**Comment by robertsLando** [comment]
@Fishwaldo Shouldn't it be cacthched here? https://github.com/OpenZWave/node-openzwave-shared/blob/master/src/callbacks.cc#L171

**Comment by Fishwaldo** [comment]
The Log Text is misleading. All Scenes are just ValueID's 

Looking at what you pointed at, thats the Notification Callback. If you have to handle scenes differently in Z2M and so on, I guess so... 

Regardless, in OZW, we see the "Scene" change and propagated it out via a ValueID. I imagine something wrong on the node-shared or your Z2M

(Remember, Scenes changed in OZW 1.6 from Notifications (like you point above) to ValueID's). Maybe thats catching you out?

**Comment by robertsLando** [comment]
> Regardless, in OZW, we see the "Scene" change and propagated it out via a ValueID. I imagine something wrong on the node-shared or your Z2M

I already have users that have scenes correctly working. When a scene change as it is a valueID I sould get a value changed notification like for other id, am I worong? 

Actually the value change event is triggered [here](https://github.com/OpenZWave/node-openzwave-shared/blob/master/src/callbacks.cc#L212) from openzwave shared and then I listen for the event in z2m like I do for all others values. I don't know why for some this works and for someone else it doesn't

**Comment by JonasPed** [comment]
Please note that it is only double activation that is not working on the device. Single activation, long hold and key released after long hold are all correctly published. Sorry if that was not clear in the original bug report. 

The device supports the following actions on each button (it contains 4 buttons). 
1. Button activated. 
2. Button released
3. Long activation of button
4. Double activation of button. 

It is only double activation that is not working correctly. On page 12 of https://youcontrol.dk/media/wysiwyg/ZHC5010-brugervejledning-Dansk.PDF you can see the scene numbers and attributes send by the device. Sorry for the manual beeing in Danish.

**Comment by robertsLando** [comment]
> Nothing wrong on OZW side: zwave2mqtt_1 | 2020-02-05 20:46:54.394 Info, Node049, Received Central Scene set from node 49: scene id=4 with key Attribute 3. Sending event notification.

@Fishwaldo So I think this one is work

@JonasPed can you confirm you don't get that log when you double press the button? I mean you don't get the message published via mqtt but even ozw doesn't detect it?

**Comment by JonasPed** [comment]
I do get that log line when double activating the button. The attached files are from when double activating the button. The one named zwave2mqtt.txt contains the console output from zwave2mqtt container and the other one is what published to MQTT.

**Comment by robertsLando** [comment]
@JonasPed Could you send the merged logs? It is really strange that you get all others events and not that one because all events are catched in the same way...

**Comment by JonasPed** [comment]
@robertsLando Not sure what you mean by "the merged logs"?

**Comment by robertsLando** [comment]
@JonasPed I mean both ozw and z2m logs together to see how z2m reacts to ozw events

**Comment by JonasPed** [comment]
@robertsLando this is what is present in zwave2mqtt.txt file. That is everything that gets printed to the console when I double activate the button.

**Comment by JonasPed** [comment]
Tried to dig a bit into this and found something strange in the ozwcache file. Below is the CommandClass element for command class 91. My understanding is that it is actually OpenZwave that generates this file. Is that correct? As you can see this does not list the "Pressed 2 Times" item element. 

			<CommandClass id="91" name="COMMAND_CLASS_CENTRAL_SCENE">
				<Compatibility />
				<State>
					<Encrypted>true</Encrypted>
					<SceneCount>4</SceneCount>
					<StaticRequests>7</StaticRequests>
				</State>
				<Instance index="1" />
				<Value type="list" genre="user" instance="1" index="1" label="Scene 1" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" vindex="0" size="0">
					<Help />
					<Item label="Inactive" value="0" />
					<Item label="Pressed 1 Time" value="1" />
					<Item label="Key Released" value="2" />
					<Item label="Key Held down" value="3" />
				</Value>
				<Value type="list" genre="user" instance="1" index="2" label="Scene 2" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" vindex="0" size="0">
					<Help />
					<Item label="Inactive" value="0" />
					<Item label="Pressed 1 Time" value="1" />
					<Item label="Key Released" value="2" />
					<Item label="Key Held down" value="3" />
				</Value>
				<Value type="list" genre="user" instance="1" index="3" label="Scene 3" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" vindex="0" size="0">
					<Help />
					<Item label="Inactive" value="0" />
					<Item label="Pressed 1 Time" value="1" />
					<Item label="Key Released" value="2" />
					<Item label="Key Held down" value="3" />
				</Value>
				<Value type="list" genre="user" instance="1" index="4" label="Scene 4" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" vindex="0" size="0">
					<Help />
					<Item label="Inactive" value="0" />
					<Item label="Pressed 1 Time" value="1" />
					<Item label="Key Released" value="2" />
					<Item label="Key Held down" value="3" />
				</Value>
				<Value type="int" genre="user" instance="1" index="256" label="Scene Count" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="-2147483648" max="2147483647" value="4">
					<Help />
				</Value>
				<Value type="int" genre="config" instance="1" index="257" label="Scene Reset Timeout" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="-2147483648" max="2147483647" value="1000">
					<Help />
				</Value>
			</CommandClass>

**Comment by robertsLando** [comment]
@JonasPed I think this is the problem @Fishwaldo

**Comment by Fishwaldo** [comment]
I need to see the full OZW_Log.txt file (or you can use ozwcp if you can’t get it from z2m). You need to remove any ozwcache file before starting. Those entries are discovered by interviewing the device and if not in the interview then it’s a big with the device).

**Comment by JonasPed** [comment]
I have around 10 of these devices in the house. After digging a bit deeper into the problem it turned out that it was only the device I used for testing that did have the issue. I have used quite some time comparing configuration of the device that did have the issue and one of the other devices. Everything under configuration was the same. After reading the manual I could see that there was one setting called "Enable Cleaning Mode" that would disable the double activation. After changing this setting everything works as expected. 

It turns out that the web GUI does not show the actual configuration. So you can not really rely on the configuration values you see in the gui - at least not for ZHC5010 devices. I have not verified with other types of devices. But I believe that is another issue and a new issue should be created. So I am closing this issue as it was only a configuration issue of the device.

**Comment by robertsLando** [comment]
With latest changes in master now you can refresh configuration paramenters by pressing a button on the ui :)

Daniel
